tests/integration/src/test/java/org/mule/test/integration/domain/tls/TlsSharedContextTestCase.java

```diff
@@ -61,27 +66,22 @@ public void localRequesterToSharedListenerUsingSharedTlsContext() throws Excepti
   }
 
   @Test
-  public void muleClientUsingLocalRequesterWithSharedTlsContextToListenerUsingSharedTlsContext() throws Exception {
-    MuleContext secondAppContext = getMuleContextForApp(SECOND_APP);
-    HttpRequesterConfig requesterConfig = secondAppContext.getRegistry().lookupObject("requestConfig");
-    HttpRequestOptions requestConfigOptions = HttpRequestOptionsBuilder.newOptions().requestConfig(requesterConfig).build();
-    testMuleClient(requestConfigOptions);
-  }
-
-  @Test
   public void muleClientUsingSharedTlsContextToListenerUsingSharedTlsContext() throws Exception {
     MuleContext domainContext = getMuleContextForDomain();
     TlsContextFactory tlsContextFactory = domainContext.getRegistry().lookupObject("sharedTlsContext2");
-    HttpRequestOptions tlsContextOptions = HttpRequestOptionsBuilder.newOptions().tlsContextFactory(tlsContextFactory).build();
-    testMuleClient(tlsContextOptions);
-  }
-
-  private void testMuleClient(HttpRequestOptions operationOptions) throws Exception {
     MuleContext context = getMuleContextForApp(SECOND_APP);
-    InternalMessage response = context.getClient().send(format("https://localhost:%s/helloAll", port3.getValue()),
-                                                        InternalMessage.builder().payload(DATA).build(), operationOptions)
-        .getRight();
-    assertThat(getPayloadAsString(response, context), is("hello all"));
+
+    HttpClient httpClient = context.getRegistry().lookupObject(HttpService.class).getClientFactory()
```

Comment:
this one i cannot put as a rule, beacuse the tlscontext can only be btained after initialization.